src/main/java/org/folio/sample/service/BookService.java

```diff
@@ -31,4 +32,10 @@ public Optional<Book> getBookById(UUID id) {
   public Book createBook(Book book) {
     return bookRepository.save(book);
   }
+  public Book updateBook(UUID id, String name, LocalDate publishedDate) {
+    Book curBook = bookRepository.getReferenceById(id);
```

Comment:
getReference is just pulling the Book from the DB that you already got in the controller. I'd recommend adding that as a parameter, rather than the ID. You could also pass the new book in, e.g. `updateBook(Book current, Book newData)`, then do the same .set and saving.

This would also move more of the logic outside the controller - in general, we want the controller's only job to be DTO conversion and routing the requests down the proper Service.